### DISCUSSION TITLE
Le pattern Stratégie

### DISCUSSION THREAD

**Discussion by giscus** [zero-filtre/zerofiltre-blog#172] [Announcements]
# Le pattern Stratégie

Veuillez indiquer ici le résume de votre article. Ex: Mettre en place un serveur de messagerie n'a jamais été aussi simple. Voici comment faire.

https://zerofiltre.tech/articles/266

**Comment by imphilippesimo** [comment]
> Pour mieux comprendre ce modèle nous allons l'utiliser dans une application de sauvegarde d'images. Supposons que nous voulions développer une fonctionnalité de notre application qui nous permet de sauvegarder les images de nos utilisateurs.   

J'ai l'impression que c'est une repétition.
Pouvons-nous nous limiter à dire : 

> Pour mieux comprendre ce modèle nous allons l'utiliser dans une application de sauvegarde d'images des utilisateurs ?

**Comment by imphilippesimo** [comment]
Je ne comprends pas très bien la fin de cet inconvénient.  

> L'application doit créer et gérer deux objets au lieu d’un seul car la plupart des cas, elle configure le contexte avec l'objet stratégie requis.  

Je ne comprends pas très bien cet avantage:
> la possibilité de permuter un algorithme à l’intérieur d’un objet à l’exécution.    

 Car j'ai l'impression que l'algortihme (l'implémentation de la stratégie à utiliser ) est défini lors de la construction du contexte et non lors de l'exécution de la stratégie.  
J'aurais dit pour cet avantage ; 

> la possibilité de changer la stratégie à utiliser sans modifier le contexte : il suffit de rentrer des paramètres différents lors de la construction du contexte.  

C'est peut être long hein 🤣, je te laisse adapter Ibrahima.

**Reply by hotiaDiallo** [reply]
J'avoue que cette partie est un tordu. Je faisais allusion au fait que "Cela augmente le nombre d'objets dans l'application" mais je me rend compte que le premier inconvénient en parle donc je l'ai enlevé.

**Comment by imphilippesimo** [comment]
> Il permet de définir une famille d’algorithmes, les mettre dans des classes séparées et de rendre leurs objets interchangeables. L'idée de ce modèle est d'encapsuler cette famille d'algorithmes dans des classes distinctes qui implémentent une interface commune.   

Moi même je n'ai pas compris cette définition avant d'avoir lu totalement l'article. 😅   
Essayons de trouver une définition que ma grand mère pourrait comprendre sans avoir à lire l'article. 😅

Je propose ceci:
> Il propose de définir un ensemble d'implémentations (des classes) d'un concept spécifique (une interface), qui seront appelées en fonction du besoin de l'appelant. 
Chaque implémentation représente une façon spécifique de résoudre le problème de l'appelant. Cela lui donne plus de flexibilité. En gros, c'est un pattern qui lui permet d'adapter la façon de faire en fonction de la situation.

**Reply by hotiaDiallo** [reply]
J'adore ta définition. Je me rend compte que j'étais trop focalisé sur les termes techniques et je me suis beaucoup répété aussi. 
Thanks.

**Reply by imphilippesimo** [reply]
Top je publie l'article et je comuunique sur Linkedin Luni ou mardi.
